webapp/channels/src/components/global_header/header_icon_button/index.ts

```diff
@@ -0,0 +1,6 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import HeaderIconButton from './header_icon_button';
```

Comment:
Nit: Can we not just export default the original function? This file seems a bit unnecessary.

Reply:
Agreed that it's a bit unnecessary, but this is a common enough pattern in our code where we either chose to move a component into its own self-contained folder to reduce the number of files in each folder or where we've removed the `mapStateToProps` from a component. At the moment, I'm more of a fan of keeping the implementation in its own file (not in an `index.tsx`), so I'm going to leave it as-is